pinot-common/src/main/java/org/apache/pinot/common/utils/DataTable.java

```diff
@@ -80,4 +85,87 @@
+    NUM_DOCS_SCANNED("numDocsScanned"),
+    NUM_ENTRIES_SCANNED_IN_FILTER("numEntriesScannedInFilter"),
+    NUM_ENTRIES_SCANNED_POST_FILTER("numEntriesScannedPostFilter"),
+    NUM_SEGMENTS_QUERIED("numSegmentsQueried"),
+    NUM_SEGMENTS_PROCESSED("numSegmentsProcessed"),
+    NUM_SEGMENTS_MATCHED("numSegmentsMatched"),
+    NUM_CONSUMING_SEGMENTS_PROCESSED("numConsumingSegmentsProcessed"),
+    MIN_CONSUMING_FRESHNESS_TIME_MS("minConsumingFreshnessTimeMs"),
+    TOTAL_DOCS("totalDocs"),
+    NUM_GROUPS_LIMIT_REACHED("numGroupsLimitReached"),
+    TIME_USED_MS("timeUsedMs"),
+    TRACE_INFO("traceInfo"),
+    REQUEST_ID("requestId"),
+    NUM_RESIZES("numResizes"),
+    RESIZE_TIME_MS("resizeTimeMs"),
+    THREAD_CPU_TIME_NS("threadCpuTimeNs"),
+    ;
+
+    private static final Map<String, MetadataKeys> _nameToEnumKeyMap = new HashMap<>();
+    // _intValueMetadataKeys contains all metadata keys which has value of int type.
+    private static final Set<MetadataKeys> _intValueMetadataKeys = ImmutableSet
+        .of(MetadataKeys.NUM_SEGMENTS_QUERIED, MetadataKeys.NUM_SEGMENTS_PROCESSED, MetadataKeys.NUM_SEGMENTS_MATCHED,
+            MetadataKeys.NUM_RESIZES, MetadataKeys.NUM_CONSUMING_SEGMENTS_PROCESSED, MetadataKeys.NUM_RESIZES);
+    // _longValueMetadataKeys contains all metadata keys which has value of long type.
+    private static final Set<MetadataKeys> _longValueMetadataKeys = ImmutableSet
+        .of(MetadataKeys.NUM_DOCS_SCANNED, MetadataKeys.NUM_ENTRIES_SCANNED_IN_FILTER,
+            MetadataKeys.NUM_ENTRIES_SCANNED_POST_FILTER, MetadataKeys.MIN_CONSUMING_FRESHNESS_TIME_MS,
+            MetadataKeys.TOTAL_DOCS, MetadataKeys.TIME_USED_MS, MetadataKeys.REQUEST_ID, MetadataKeys.RESIZE_TIME_MS,
+            MetadataKeys.THREAD_CPU_TIME_NS);
+    private final String _name;
+
+    MetadataKeys(String name) {
+      this._name = name;
+    }
+
+    // getByOrdinal returns an optional enum key for a given ordinal
+    public static Optional<MetadataKeys> getByOrdinal(int ordinal) {
+      if (ordinal >= MetadataKeys.values().length) {
+        return Optional.empty();
+      }
+      return Optional.ofNullable(MetadataKeys.values()[ordinal]);
+    }
+
+    // getByName returns an optional enum key for a given name.
+    public static Optional<MetadataKeys> getByName(String name) {
+      return Optional.ofNullable(_nameToEnumKeyMap.getOrDefault(name, null));
+    }
+
+    // isIntValueMetadataKey returns true if the given key has value of int type.
+    public static boolean isIntValueMetadataKey(MetadataKeys key) {
+      return _intValueMetadataKeys.contains(key);
+    }
+
+    // isLongValueMetadataKey returns true if the given key has value of long type.
+    public static boolean isLongValueMetadataKey(MetadataKeys key) {
+      return _longValueMetadataKeys.contains(key);
+    }
+
+    // getName returns the associated name(string) of the enum key.
+    public String getName() {
+      return _name;
+    }
+
+    static {
```

Comment:
Put this block following the map definition for better readability

Reply:
Oh, the code block putting here was conduct by IntellJ reformat. I'd suggest keep as it is, since later someone change this file and run IntellJ reformatting before commit, it will be moved to here anyway.

Reply:
The code was put here by Intellj reformatting. I'd suggest keep it here, since assume some change this file, and run IntellJ reformat before commit, it will be moved to here anyway.